docs/advanced_topics/customisation/page_editing_interface.rst

```diff
@@ -61,6 +61,49 @@ However, template output from :class:`~wagtail.wagtailcore.fields.RichTextField`
 
 If you're interested in extending the capabilities of the Wagtail WYSIWYG editor (``hallo.js``), See :ref:`extending_wysiwyg`.
 
+.. _custom_config:
+
+Passing a custom ``hallo.js`` editor configuration
+--------------------------------------------------
+
+A custom ``hallo.js`` JSON configuration can be passed through the default widgets RichTextArea ``editor_config`` attribute.
+
+.. code-block:: python
+
+    from wagtail.wagtailcore.fields import RichTextField
+    from wagtail.wagtailadmin.edit_handlers import FieldPanel
+
+    from wagtail.wagtailcore.fields import RichTextArea
+
+    simple_config={
+        'halloheadings': {
```

Comment:
I would be keen to pursue passing a whole configuration and keeping the **editor_config** as the attribute name. Only passing **plugins** seems very biased towards the hallo editor and would possible not fit when implementing other editors.

I would agree with you suggestions of updating the configuration to reflect/allow all possible configuration options of an editor to be passed. 

I will also update page-editor.js handling the passed code later today.